src/pages/RouteView.tsx

```diff
@@ -0,0 +1,141 @@
+import { NavBar } from '../components/NavigationBar';
+import { createSearchParams, useNavigate, useSearchParams } from 'react-router-dom';
+import { getCurrentUser, getRouteById } from '../xplat/api';
+import { useQuery } from 'react-query';
+import { NaturalRules, Route, RouteStatus, User, UserStatus, invalidateDocRefId } from '../xplat/types';
+import { queryClient } from '../App';
+import placeholder_image from '../placeholder_image.jpg';
+import { useEffect, useState } from 'react';
+import { buildUserByIDFetcher } from '../utils/queries';
+
+const RouteView = () => {
+  const [params] = useSearchParams();
+  const navigate = useNavigate();
+  const [hasUserUID, setHasUserUID] = useState<boolean>(false);
+  const [userUID, setUserUID] = useState<string>('');
+
+  const hasRouteUID: boolean = params.has('uid');
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const routeUID: string = (hasRouteUID ? params.get('uid')! : '');
+
+  const routeQuery = useQuery(
+    ['routes', routeUID],
+    Route.buildFetcherFromDocRefId(routeUID),
+    {
+      enabled: hasRouteUID,
+    }
+  );
+
+  useEffect(() => {
+    const fetchUserUID = async () => {
+      getCurrentUser().then((user: User) => {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        setUserUID(user.docRef!.id);
+        setHasUserUID(true);
+      });
+    };
+    // only refetch data if we have not set userID before
+    if (!hasUserUID) {
+      fetchUserUID();
+    }
+  }, []);
+
+  // TODO: use xplat user fetcher once it gets updated
+  const userQuery = useQuery(
+    ['currentUser', userUID],
+    buildUserByIDFetcher(userUID),
+    {
+      enabled: hasUserUID,
+    }
+  );
+
+  if (!hasRouteUID) {
+    // with our navigation this should never happen, but can never be too safe?
+    console.error('no uid params given');
+    return null;
+  }
+
+  if (!hasUserUID || routeQuery.isLoading || userQuery.isLoading) {
+    return null;
+  }
+
+  if (routeQuery.isError || routeQuery.data === undefined) {
+    console.error(routeQuery.error);
+    return null;
+  }
+
+  if (userQuery.isError || userQuery.data === undefined) {
+    console.error(userQuery.error);
+    return null;
+  }
+
+  const navToRouteFeed = () => {
+    navigate({
+      pathname: '/routefeed',
+      search: `?${createSearchParams({ uid: routeUID })}`
+    });
+  };
+
+  const archiveThisRoute = async () => {
+    getRouteById(routeUID).upgradeStatus().then(() => {
+      // important to invalidate doc ref id first before invalidating the query
+      // so it can run with knowledge of invalidated doc ref id
+      invalidateDocRefId(routeUID);
+      queryClient.invalidateQueries({ queryKey: ['routes', routeUID] });
+    });
+  };
+
+  const notAssigned = 'not assigned';
+  const displayNaturalRules = routeQuery.data.naturalRules ? NaturalRules[routeQuery.data.naturalRules] : notAssigned;
```

Comment:
It's up to you but I would rather this logic be directly within the JSX rather than separate.